validate/validate.go

```diff
@@ -179,13 +199,49 @@ func minVersionCheck(quote *pb.QuoteV4, opts *Options) error {
 		return fmt.Errorf("TEE TCB security-version number %d is less than the required minimum %d",
 			quote.GetTdQuoteBody().GetTeeTcbSvn(), opts.TdQuoteBodyOptions.MinimumTeeTcbSvn)
 	}
-	if !isSvnHigherOrEqual(quote.GetHeader().GetQeSvn(), opts.HeaderOptions.MinimumQeSvn) {
+	qeSvn := uint32(binary.LittleEndian.Uint16(quote.GetHeader().GetQeSvn()))
+	pceSvn := uint32(binary.LittleEndian.Uint16(quote.GetHeader().GetPceSvn()))
+	if qeSvn < opts.HeaderOptions.MinimumQeSvn {
 		return fmt.Errorf("QE security-version number %d is less than the required minimum %d",
-			quote.GetHeader().GetQeSvn(), opts.HeaderOptions.MinimumQeSvn)
+			qeSvn, opts.HeaderOptions.MinimumQeSvn)
 	}
-	if !isSvnHigherOrEqual(quote.GetHeader().GetPceSvn(), opts.HeaderOptions.MinimumPceSvn) {
+	if pceSvn < opts.HeaderOptions.MinimumPceSvn {
 		return fmt.Errorf("PCE security-version number %d is less than the required minimum %d",
-			quote.GetHeader().GetPceSvn(), opts.HeaderOptions.MinimumPceSvn)
+			pceSvn, opts.HeaderOptions.MinimumPceSvn)
+	}
+	return nil
+}
+
+func validateXfam(value []byte, fixed1, fixed0 uint64) error {
+	if len(value) == 0 {
+		return nil
+	}
+	if len(value) != abi.XfamSize {
+		return fmt.Errorf("xfam size is invalid")
+	}
+	xfam := binary.LittleEndian.Uint64(value[:])
+	if xfam&fixed1 != fixed1 {
+		return fmt.Errorf("unauthorized xfam as xfamFixed1 bits are unset")
```

Comment:
these error messages reference an internal constant. You may want to state what mask is required vs what mask is allowed.

Reply:
Changed the error message.